snippets/scripts/jira-components-to-compass-components/migrateJiraComponentsToCompassComponents.py

```diff
@@ -0,0 +1,302 @@
+        issues = response['issues']
+        total = response['total']
+        start_at += len(issues)
+        all_issues.extend(issues)
+        if start_at >= total:
+            break
+        url = f"/rest/api/3/search?{urllib.parse.urlencode(params)}&startAt={start_at}"
+
+    # the jql api will return issues by component name
+    # which can include compass components with same name as well
+    # we need to filter out the ones that has the current Jira component in them
+    all_issues = [issue for issue in all_issues if any(component_id in component['id'] for component in issue['fields']['components'])]
+
+    return all_issues
+
+
+def get_tenant_info():
+    response = make_gql_api_call(TENANT_INFO_QUERY, {"hostNames": [DOMAIN_NAME]})
+    return response['data']['tenantContexts'][0]['cloudId']
+
+
+def create_compass_component(cloud_id,component):
+    url = f"/gateway/api/graphql"
+    headers = {
+        'Content-Type': 'application/json'
+    }
+    variables = {
+        "cloudId": cloud_id,
+        "input": {
+            "name": component['name'],
+            "typeId": DEFAULT_COMPASS_COMPONENT_TYPE
+        }
+    }
+
+    response = make_gql_api_call(COMPASS_CREATE_COMPONENT_MUTATION, variables)
+    return response['data']['compass']['createComponent']
+
+
+# at least one project must have compass ON
+def find_compass_component_in_jira(component_name):
+    url = f"/rest/api/3/component?query={component_name}"
+    response = make_api_call(url, "GET")
+    if len(response['values']) > 0 and 'ari' in response['values'][0]['ari']:
+        return response['values'][0]
+    return None
+
+
+def update_issue_components(issue, jira_component_id_to_remove, compass_component_id_to_add):
+    url = f"/rest/api/3/issue/{issue['key']}"
+    # remove the Jira component and add the compass component
+    existing_components = issue['fields']['components']
+    components_after = [{'id': component['id']} for component in existing_components if component['id'] != jira_component_id_to_remove]
+    components_after.append({'id': compass_component_id_to_add})
+
+    data = {
+        "update": {
+            "components": [
+                {'set': components_after},
+            ]
+        }
+    }
+
+    make_api_call(url, "PUT", data, no_response=True)
+
+
+def is_valid_project(project_key):
+    url = f"/rest/api/3/project/{project_key}"
+    response = make_api_call(url, "GET")
+    if response is None:
+        return False
+    if response['projectTypeKey'] == "software" and response['style'] == "classic":
+        return True
+    return False
+
+
+def does_project_have_compass_toggle_on():
+    try:
+        url = f"/rest/api/3/project/{PROJECT_KEY}/properties/jira.global.components"
+        response = make_api_call(url, "GET")
+        if response is None:
+            return False
+        return response['value']
+    except Exception as e:
+        return False
+
+
+def main():
+    if not is_valid_project(PROJECT_KEY):
```

Comment:
Seems like this will swallow all errors and output the below text regardless of error type. I was typing the wrong email and it failed auth, so the error message was a bit misleading. 😬

Reply:
alright added a check for credentials